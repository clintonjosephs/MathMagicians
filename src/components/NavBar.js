import React from 'react';
import { NavLink } from 'react-router-dom';
import styles from './NavBar.module.css';
import logo from './calculator.svg';

const NavBar = () => {
  const links = [
    {
      id: 1,
      path: '/',
      text: 'Home',
    },
    {
      id: 2,
      path: '/calculator',
      text: 'Calculator',
    },
    {
      id: 3,
      path: '/quotes',
      text: 'Quotes',
    },
  ];

  return (
    <nav className={styles.navBar}>
      <h2>
        <img src={logo} alt="Logo" style={{ width: '20px' }} />
        {' '}
        Math-Magicians
      </h2>
      <ul>
        {links.map(({ id, path, text }) => (
          <li key={id}>
            <NavLink
              to={path}
              className="link"
              style={({ isActive }) => ({
                textDecoration: 'none',
                margin: '0 10px',
                borderBottom: isActive ? 'solid 5px #f5913e' : '',
                color: isActive ? '#f5913e' : 'black',
              })}
            >
              {text}
            </NavLink>
          </li>
        ))}
      </ul>
    </nav>
  );
};

export default NavBar;
