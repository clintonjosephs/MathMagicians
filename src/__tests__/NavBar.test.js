import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import NavBar from '../components/NavBar';
import { BrowserRouter as Router } from 'react-router-dom';

describe('Use Jest snapshots to test the navbar.', () => {
  it('Renders NavBar component', () => {
    const component = renderer.create(
      <Router>
        <NavBar />
      </Router>
    );
    expect(component).toMatchSnapshot();
  });
});

describe('Test user operation on NavLink', () => {
  it('Checks how links toggle active on user click', () => {
    render(
      <Router>
        <NavBar />
      </Router>
    );
    expect(screen.getByRole('list').childElementCount).toBe(3);

    userEvent.click(screen.getByText(/quotes/i));

    expect(screen.getByText('Quotes').closest('a')).toHaveAttribute(
      'class',
      'link active'
    );

    userEvent.click(screen.getByText(/home/i));

    expect(screen.getByText('Quotes').closest('a')).not.toHaveAttribute(
      'class',
      'link active'
    );

    expect(screen.getByText('Home').closest('a')).toHaveAttribute(
      'class',
      'link active'
    );
  });
});
