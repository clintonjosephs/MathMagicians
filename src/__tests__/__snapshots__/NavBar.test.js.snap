// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Use Jest snapshots to test the navbar. Renders NavBar component 1`] = `
<nav
  className="navBar"
>
  <h2>
    <img
      alt="Logo"
      src="calculator.svg"
      style={
        Object {
          "width": "20px",
        }
      }
    />
     
    Math-Magicians
  </h2>
  <ul>
    <li>
      <a
        aria-current="page"
        className="link active"
        href="/"
        onClick={[Function]}
        style={
          Object {
            "borderBottom": "solid 5px #f5913e",
            "color": "#f5913e",
            "margin": "0 10px",
            "textDecoration": "none",
          }
        }
      >
        Home
      </a>
    </li>
    <li>
      <a
        className="link"
        href="/calculator"
        onClick={[Function]}
        style={
          Object {
            "borderBottom": "",
            "color": "black",
            "margin": "0 10px",
            "textDecoration": "none",
          }
        }
      >
        Calculator
      </a>
    </li>
    <li>
      <a
        className="link"
        href="/quotes"
        onClick={[Function]}
        style={
          Object {
            "borderBottom": "",
            "color": "black",
            "margin": "0 10px",
            "textDecoration": "none",
          }
        }
      >
        Quotes
      </a>
    </li>
  </ul>
</nav>
`;
